### ISSUE TITLE
MSW: Callback function does not have async

### ISSUE THREAD

**Issue by tatsuya-asami** [orval-labs/orval#1582]
## What are the steps to reproduce this issue?

1.  Set output like this
```
output: {
    mock: {
       type: 'msw',
       delay: false,
       useExamples: true
    },
}
```
2. Execute `npx orval`

## What happens?
Most msw functions have type error about `'await' expressions are only allowed within async functions and at the top levels of modules.ts`

```ts
export const getUserControllerRemoveMockHandler = (overrideResponse?: void | ((info: Parameters<Parameters<typeof http.delete>[1]>[0]) => Promise<void> | void)) => {
  return http.delete('*/user',  (info) => { // need async this line
  if (typeof overrideResponse === 'function') {await overrideResponse(info); }
    return new HttpResponse(null,
      { status: 200,
        
      })
  })
}
```

## What were you expecting to happen?
All functions that have await should have async
```ts
export const getUserControllerRemoveMockHandler = (overrideResponse?: void | ((info: Parameters<Parameters<typeof http.delete>[1]>[0]) => Promise<void> | void)) => {
  return http.delete('*/user',  async (info) => { // add async here
  if (typeof overrideResponse === 'function') {await overrideResponse(info); }
    return new HttpResponse(null,
      { status: 200,
        
      })
  })
}
```

## Any logs, error output, etc?
```ts
export const getUserControllerRemoveMockHandler = (overrideResponse?: void | ((info: Parameters<Parameters<typeof http.delete>[1]>[0]) => Promise<void> | void)) => {
  return http.delete('*/user',  (info) => { // need async this line
  if (typeof overrideResponse === 'function') {await overrideResponse(info); }
    return new HttpResponse(null,
      { status: 200,
        
      })
  })
}
```

## Any other comments?
Probably, it will happen when response type is void.
This issue did not happen when I use orval 6.31.0

## What versions are you using?
```
  System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 70.23 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  npmPackages:
    axios: ^1.7.4 => 1.7.4 
    msw: ^2.3.5 => 2.3.5 
    orval: ^7.0.1 => 7.0.1 
    react: 19.0.0-rc-19bd26be-20240815 => 19.0.0-rc-19bd26be-20240815 
    swr: ^2.2.5 => 2.2.5 
```

**Comment by soartec-lab** [comment]
@tatsuya-asami 

Thank you for made this report.
This issue has already been fixed in #1567 
So you can try latest version 👍

**Comment by tatsuya-asami** [comment]
> 
> Thank you for made this report. This issue has already been fixed in #1567 So you can try latest version 👍

@soartec-lab 
Thank you so much! I will try the latest version❤️
